airflow/kubernetes/pod_generator.py

```diff
@@ -157,6 +136,10 @@ def gen_pod(self) -> k8s.V1Pod:
     @staticmethod
     def add_xcom_sidecar(pod: k8s.V1Pod) -> k8s.V1Pod:
         """Adds sidecar"""
+        warnings.warn(
+            "This function is deprecated. "
+            "Please use airflow.providers.cncf.kubernetes.utils.xcom_sidecar.add_xcom_sidecar instead"
```

Comment:
Worth adding a note in Updating.md too please

Reply:
added